### PR TITLE
rank issues in sprints and backlog

### PR DESCRIPTION
**Pull request by bic-marz**
after other changes are merged we have to sort the issues in the sprints when these get fetched, because as of now they are displayed in reversed order

### CODE REVIEW

File: project-canvas/src/components/backlog-view/BacklogView.tsx [hunk 1]
```diff
@@ -58,10 +59,14 @@ export function BacklogView() {
...
-                issue.type !== "Epic" && issue.type !== "Subtask"
-            ),
+            issues: issues
+              .filter(
+                (issue: Issue) =>
+                  issue.type !== "Epic" && issue.type !== "Subtask"
+              )
+              .sort((issueA: Issue, issueB: Issue) =>
+                sortIssuesByRank(issueA, issueB)
+              ),
```
Comment (IhsenBouallegue) [inline project-canvas/src/components/backlog-view/BacklogView.tsx:69]: Can we do this for backlog too

File: project-canvas/src/components/backlog-view/helpers/backlogHelpers.ts [hunk 2]
```diff
@@ -20,3 +20,13 @@ export const sortSprintsByActive = (sprintA: Sprint, sprintB: Sprint) => {
...
 }
+
+export const sortIssuesByRank = (issueA: Issue, issueB: Issue) => {
+  if (issueA.rank < issueB.rank) {
+    return -1
+  }
+  if (issueA.rank > issueB.rank) {
+    return 1
+  }
+  return 0
```
Comment (IhsenBouallegue) [inline project-canvas/src/components/backlog-view/helpers/backlogHelpers.ts:31]:
```suggestion
  return issueA.rank.localeCompare(issueB.rank)
```

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts [hunk 3]
```diff
@@ -181,13 +181,13 @@ class JiraCloudProvider implements ProviderApi {
   }
 
   async fetchIssues(url: string): Promise<Issue[]> {
+    const rankCustomField = this.customFields.get("Rank")
```
Comment (IhsenBouallegue) [inline project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:184]:
```suggestion
    const rankCustomField = this.customFields.get("Rank") || ""
```

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts [hunk 4]
```diff
@@ -200,15 +200,28 @@ class JiraCloudProvider implements ProviderApi {
         storyPointsEstimate: await this.getIssueStoryPointsEstimate(
           element.key
         ),
+        rank: element.fields[rankCustomField!],
```
Comment (IhsenBouallegue) [inline project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:203]:
```suggestion
        rank: element.fields[rankCustomField],
```

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts [hunk 5]
```diff
@@ -200,15 +200,28 @@ class JiraCloudProvider implements ProviderApi {
...
+    issue: string,
+    rankBefore: string,
+    rankAfter: string
+  ): Promise<void> {
     return new Promise((resolve, reject) => {
+      const rankCustomField = this.customFields.get("Rank")
+      const body = {
+        rankCustomFieldId: rankCustomField!.match(/_(\d+)/)![1],
+        issues: [issue],
+        ...(rankAfter ? { rankAfterIssue: rankAfter } : {}),
```
Comment (IhsenBouallegue) [inline project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:222]:
```suggestion
        ...(rankAfter ?? { rankAfterIssue: rankAfter }),
```
not sure check if it works

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts [hunk 6]
```diff
@@ -218,16 +231,13 @@ class JiraCloudProvider implements ProviderApi {
...
-              "${issue}"
-            ]
-          }`,
+          body: JSON.stringify(body),
         }
       )
         .then(() => {
           resolve()
         })
+
```
Comment (IhsenBouallegue) [inline project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:240]: remove blank line

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts [hunk 7]
```diff
@@ -218,16 +231,13 @@ class JiraCloudProvider implements ProviderApi {
...
-            "issues": [
-              "${issue}"
-            ]
-          }`,
+          body: JSON.stringify(body),
         }
       )
         .then(() => {
           resolve()
         })
```
Comment (IhsenBouallegue) [inline project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:239]:
```suggestion
        .then(() => resolve())
```

File: project-extender/src/types/jira.ts [hunk 8]
```diff
@@ -23,5 +23,6 @@ export interface JiraIssue {
     status: { name: string }
     issuetype: { name: string }
     customfield_10107: number
+    [rankCustomField: string]: string | unknown
```
Comment (IhsenBouallegue) [inline project-extender/src/types/jira.ts:26]: let's mark this as todo to improve it

**Review by IhsenBouallegue — APPROVED** [review]

File: project-canvas/src/components/backlog-view/BacklogView.tsx (hunk 1 above)
Comment (bic-marz) [reply project-canvas/src/components/backlog-view/BacklogView.tsx:69]: done

File: project-canvas/src/components/backlog-view/helpers/backlogHelpers.ts (hunk 2 above)
Comment (bic-marz) [reply project-canvas/src/components/backlog-view/helpers/backlogHelpers.ts:31]: yas queen

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts (hunk 3 above)
Comment (bic-marz) [reply project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:184]: yaas

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts (hunk 4 above)
Comment (bic-marz) [reply project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:203]: qween

File: project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts (hunk 5 above)
Comment (bic-marz) [reply project-extender/src/providers/jira-cloud-provider/JiraCloudProvider.ts:222]: image

File: project-extender/src/types/jira.ts (hunk 8 above)
Comment (bic-marz) [reply project-extender/src/types/jira.ts:26]: dan

**Pull request merged**